src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -176,108 +196,14 @@ public Uri GetEffectivePath(string commandSpecifiedPath)
             return builder.Uri;
         }
 
-        public Uri GetEffectivePath(IReadOnlyList<string> sections, bool requiresBody, out int filePathIndex)
+        public Uri GetEffectivePathForPrompt()
```

Comment:
@tlmii , I looked at old implementation and this one. I get the refactoring of first GetEffectivePath(...). Why did we completely remove the second one?

Reply:
The second one (`GetEffectivePath(IReadOnlyList<string>, bool, out int)`) was basically just a special-case of the first one used to create the REPL prompt string. It was mostly duplicated code. Once I went through and commented what it was doing, I realized it wasn't even really necessary. So the new function (`GetEffectivePathForPrompt()`) just handles some of the special casing first and calls into the other one.

Reply:
Thanks for the explanation Tim!